### DISCUSSION TITLE
AuthApiError: Error sending confirmation mail

### DISCUSSION THREAD

**Discussion by psiaux** [supabase/supabase#37030] [Questions]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug
Sometimes, the sign up function returns the following error:
`AuthApiError: Error sending confirmation mail
      at handleError (webpack-internal:///(action-browser)/./node_modules/.pnpm/@supabase+gotrue-js@2.62.0/node_modules/@supabase/gotrue-js/dist/module/lib/fetch.js:46:11)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async _handleRequest (webpack-internal:///(action-browser)/./node_modules/.pnpm/@supabase+gotrue-js@2.62.0/node_modules/@supabase/gotrue-js/dist/module/lib/fetch.js:93:9)
      at async _request (webpack-internal:///(action-browser)/./node_modules/.pnpm/@supabase+gotrue-js@2.62.0/node_modules/@supabase/gotrue-js/dist/module/lib/fetch.js:73:18)
      at async SupabaseAuthClient.signUp (webpack-internal:///(action-browser)/./node_modules/.pnpm/@supabase+gotrue-js@2.62.0/node_modules/@supabase/gotrue-js/dist/module/GoTrueClient.js:236:23)
      at async signUp (webpack-internal:///(action-browser)/./src/app/cadastro/_actions/index.ts:26:17)
      at async project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\compiled\next-server\app-page.runtime.dev.js:39:406
      at async t4 (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\compiled\next-server\app-page.runtime.dev.js:38:6379)
      at async rk (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\compiled\next-server\app-page.runtime.dev.js:39:25934)
      at async doRender (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:1394:30)
      at async cacheEntry.responseCache.get.routeKind (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:1555:28)
      at async DevServer.renderToResponseWithComponentsImpl (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:1463:28)
      at async DevServer.renderPageComponent (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:1856:24)
      at async DevServer.renderToResponseImpl (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:1894:32)
      at async DevServer.pipeImpl (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:911:25)
      at async NextNodeServer.handleCatchallRenderRequest (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\next-server.js:271:17)
      at async DevServer.handleRequestImpl (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\base-server.js:807:17)
      at async project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\dev\next-dev-server.js:331:20
      at async Span.traceAsyncFn (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\trace\trace.js:151:20)
      at async DevServer.handleRequest (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\dev\next-dev-server.js:328:24)
      at async invokeRender (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\router-server.js:163:21)
      at async handleRequest (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\router-server.js:342:24)
      at async requestHandlerImpl (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\router-server.js:366:13)
      at async Server.requestListener (project_folder\node_modules\.pnpm\next@14.1.0_@babel+core@7.23.7_react-dom@18.2.0_react@18.2.0\node_modules\next\dist\server\lib\start-server.js:140:13) {
    __isAuthError: true,
    status: 500
  }`

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

`const res = await supabase.auth.signUp({ email, password })`
 
`if (res.error !== null) console.log({ signup_error: res.error })`

## Expected behavior

`res.error` should be `null`.

## System information

- OS: Windows 10
- Browser Brave
- Version of @supabase/ssr: ^0.0.10
- Version of Node.js: v20.9.0

## Additional context

I'm using Next.js and Supabase's integration with Resend (as custom SMTP). It happens both on development and production, although in production there's less trace log information.

**Comment by vladaman** [comment]
We have same issue

**Comment by encima** [comment]
Thanks for opening! How often is this occurring? Is it upon every signup?

**Comment by ghost** [comment]
# i'v got the same issue @ every signup

**auth/v1/otp 500 (Internal Server Error)**

```
AuthApiError: Error sending confirmation mail
    at handleError2 (@supabase_supabase-js.js?v=f9aeffef:4096:9)
    at async _handleRequest2 (@supabase_supabase-js.js?v=f9aeffef:4131:5)
    at async _request (@supabase_supabase-js.js?v=f9aeffef:4118:16)
    at async SupabaseAuthClient.signInWithOtp (@supabase_supabase-js.js?v=f9aeffef:4991:27)
```

**Comment by komiljonovshohjahon** [comment]
Got the same issue when using custom smtp (brevo)

**Comment by xbdh** [comment]
do not use stmp server  ，just paid。 this error cost me a day to debug . try to delete this project and  create new

**Comment by paulfilipan23** [comment]
i've had the same issue.
I did the following, confirmed the domain, i'm using brevo.
restarted the project. 
The last one kinda fixed

**Comment by psiaux** [comment]
> Thanks for opening! How often is this occurring? Is it upon every signup?

Not every, but a lot of them. Supabase support basically told me to try another service (I've used the official integration of Resend and Supabase).

**Comment by rxnveer** [comment]
Suddenly started facing this issue today... restarting the project didn't make a difference. Any one got a fix? @psiaux did you just stop using resend?

Update: switching to postmark fixed it...

**Comment by Mubashar639** [comment]
<img width="853" alt="image" src="https://github.com/supabase/supabase/assets/42145875/2ba2de41-b400-4db6-bd72-d5d2a07b53a5">
It might be port issue, try to change the port, I fixed it by 587 port

**Comment by eluchsinger** [comment]
Hey
same problem here. With Resend and also with other providers.

Also tried changing port to 587 or restarting the project. Nothing helped so far...

**Comment by kawaneao1996** [comment]
I menshioned it by mistake.
Don't care about it.

**Comment by techsin** [comment]
same error using AWS SES, SMTP server works, tried with different email client, i get emails through that, so supabase should be able to send emails with the same configuration. Is it some tls issue?

**Comment by alissanguyen** [comment]
@psiaux @rxnveer @eluchsinger same issue here, provider is resend. Followed their official docs as well, any of you were able to fix it? TIA.

**Comment by roobox** [comment]
Any updates on this? I am running a production application with Supabase and face the same issue. This is highly critical for me, as users can not register occasionally!

**Comment by rxnveer** [comment]
Hey @eluchsinger @alissanguyen @roobox, I haven't seen or had any users report this issue again since my original comment, but I also don't have much logging set up and the Resend dashboard never reports any errors, so I guess it could still be happening occasionally.

This is what Resend told me after I reported the issue in March:

<img width="1197" alt="Screenshot 2024-05-12 at 19 20 17" src="https://github.com/supabase/supabase/assets/50057275/6ae8f3a7-d19f-41b9-b683-83d1d13388d4">

It might be worth raising a ticket if you haven't done so yet?

Temporarily switching to Postmark did work for me previously, but I'd rather stay with Resend (if its reliability can be guaranteed...)

**Comment by AshutoshDash1999** [comment]
> i've had the same issue. I did the following, confirmed the domain, i'm using brevo. restarted the project. The last one kinda fixed

@paulfilipan23 how to confirm domain?

**Comment by BFMarks** [comment]
same issue - this is more time consuming than building from scratch

**Comment by JoaquimLey** [comment]
Randomly getting this issue on all of my production apps!

Kinda bad timing since we were having some paid acquisition going :|

**Comment by tommasodelorenzo** [comment]
Same error here in production! HIGHLY CRITICAL. No user can register or login via magic link!!!
Using resend with configuration as in documentation.

**Comment by JoaquimLey** [comment]
(not a solution)

I just tested disabling my custom SMTP (resend) and it works but only for 3 users/hour 😅

Re-enabling resurfaced the issue and the problem I'm assuming is not on the resend side as I don't have any logs there and it throws on the supabse's Auth client (looking at supabase logs).

**Comment by tommasodelorenzo** [comment]
> (not a solution)
> 
> I just tested disabling my custom SMTP (resend) and it works but only for 3 users/hour 😅
> 
> Re-enabling resurfaced the issue and the problem I'm assuming is not on the resend side as I don't have any logs there and it throws on the supabse's Auth client (looking at supabase logs).

Yes the same here. It works disabling custom SMTP (resend). But this is not clearly a solution given that supabase built-in smtp is rate limited

**Comment by rxnveer** [comment]
Looks like the issues today were due to Resend having an outage... https://resend-status.com/incident/386050

**Comment by eichgi** [comment]
As of now the issue remains...

**Comment by itsmikesharescode** [comment]
any update in this ?

**Comment by itsmikesharescode** [comment]
asdasdas yerksss not working

**Comment by eichgi** [comment]
I created a ticket in supabase to ask for more details regarding resend and supabase integration.

They mentioned the following:

`"error":"gomail: could not send email 1: 450 The [yourdomain.com](http://yourdomain.com/) domain is not verified. Please, add and verify your domain on https://resend.com/domains"`

And this is exactly what I did, added a domain to my resend account along all the steps.

Voìla! It start sending emails correctly from supabase by setting up the proper smtp resend credentials.

My 2 cents here.

**Comment by GiorgiMazm** [comment]
i have the same issue with SendGrid, does anyone know how to fix?

**Comment by jellohouse** [comment]
Same issue for me today - out of the blue i see this error "Error sending magic link email"

Also using Resend as SMTP in Supabase project settings....

**Comment by Vinitpal** [comment]
was working just fine for me in last week, now getting same error

**Comment by andupotorac** [comment]
Same here - Resend integration, using subdomain, getting this:
{"code":500,"error_code":"unexpected_failure","msg":"Error sending confirmation email","error_id":"8cabf694520e6677-MAD"}%

**Comment by ianxxiao** [comment]
same problem

raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Server error '500 Internal Server Error' for url 'https://ncohqoomsvifclkclnfk.supabase.co/auth/v1/signup'

gotrue.errors.AuthApiError: Error sending confirmation email

**Comment by jellohouse** [comment]
For me it turned out to actually be a problem with Resend.

I went to the "Logs" section in Resend and on the failed emails it said something like `Configuration set <xxxxxxxxxx> does not exist.`

But i'm still not sure what caused this, i contacted them but they were slow to reply. So i just deleted my domain in Resend and added it again and the error went away.

**Comment by igornowosad** [comment]
In my situation (self hosted supabase client with Mailtrap email devliery platform) with the same error I've checked the auth logs (Logs -> Auth) and I see that there's an error
```
    "error": "Error sending confirmation email: gomail: could not send email 1: 554 5.7.1 Demo domains can only be used to send emails to account owners. You can only send testing emails to your own email address.",
```
So as the message says - it's a problem caused by Mailtrap 😄

**Comment by Paul-boxpower** [comment]
Error [AuthApiError: Error sending confirmation email], i am using resend and i followed their documentation

**Comment by ShivamGupta-SM** [comment]
I am using brevo still the problem remains the same. Anyone figured out how to do changes ?

**Comment by bartaldavid** [comment]
I checked the API key in Resend, when I clicked on the "last used" cell, I could get to its logs. It said:
`{
    "name": "validation_error",
    "message": "The associated domain with your API key is not verified. Please, create a new API key with full access or with a verified domain.",
    "statusCode": 400
}`

So what I did is I created a full access api key and replaced the old password in Supabase with that. The problem was solved. The issue for us probably was that we used a "Sending access" api key.

**Comment by johnnykoo84** [comment]
For the previous project, it works fine
for my 2nd project, I get the same issue here, and full access does not help me either. Anybody at supabase is working on this please?

**Comment by rizkraf** [comment]
Maybe you can change your sender email with a domain that verifies in your email services, in my case I use Resend, and that fixed it

![image](https://github.com/user-attachments/assets/9abc3c52-bacd-482c-ab1f-de0524de7eca)

![image](https://github.com/user-attachments/assets/6678c2b8-0eb7-4b73-ab72-f95dbdcc1680)

**Comment by haadbaig** [comment]
I faced the same issue with Brevo as custom smtp. I just regenerated the login and master password in brevo and used it and it started working.

**Comment by julianeden** [comment]
Same issue with Resender. Nowhere I can find to see logs to debug the issue. Not showing up in resender logs and idk if/where supabase has email logs.

**Comment by andrilla-francis** [comment]
I haven't had any issues using Supabase with the Resend integration until just yesterday—been using it for over a year now. I will point out—since no one else has—turning off the SMTP integration doesn't make a difference. I still get the same errors.

I wonder if anyone else is using custom HTML in the email? Could this be related, though it's never been an issue for me before. When I change the content of the email to just `{{ .ConfirmationURL }}` for example, it works correctly.

**Comment by eldominics** [comment]
you can just get the error message in details from supabase logs.. look into it and you will most likely see the error. you can also copy the entire body of the error message and feed it into chatGPT, it should be able to give you an explanation as to why you are getting the error.

**Comment by Hallidayo** [comment]
Hi all, just checking in here, is this still an issue for users?

**Comment by jamiechong** [comment]
If you're using Amazon SES, you need to ensure you're SES is not in "sandbox mode" or that you're sending to an email that is approved to work while in sandbox mode.

**Comment by walifile** [comment]
![Image](https://github.com/user-attachments/assets/4ae43d46-8ff6-4860-bee8-5ae96563d1e1)

in my case your sender email must be verified email example@your_verified_domain.com

**Comment by ArtiomXD** [comment]
> Maybe you can change your sender email with a domain that verifies in your email services, in my case I use Resend, and that fixed it
> 
> ![image](https://github.com/user-attachments/assets/9abc3c52-bacd-482c-ab1f-de0524de7eca)
> 
> ![image](https://github.com/user-attachments/assets/6678c2b8-0eb7-4b73-ab72-f95dbdcc1680)

thanks, helped me

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.

**Comment by coreAux** [comment]
I was having issues with this as well and @ArtiomXD's answer was the solution for me.

May I suggest that an explicit explanation that the sender e-mail domain has to be the same as the verified domain could be added to [the docs](https://resend.com/docs/send-with-supabase-smtp)? Because it's not stated there at the moment.

**Comment by MikeLundahl** [comment]
I just switched over to Resend / Supabase integration (was using Sendgrid previously), and I'm having the same issue. I'm just getting 450 error code and no more information than that in the logs in Supabase. No logs of usage at all in Resend. 

I have a verified custom domain that has worked with no issues using Sendgrid.

**Reply by coreAux** [reply]
Is the domain you set as sender the same as the verified domain?
I.e.:
```
 verified domain: mydomain.com
 sender email: hello@mydomain.com
```

**Reply by MikeLundahl** [reply]
Yes.. But I did also set the verified domain under a subdomain as recommended by Resend. Could this be the issue?

**Reply by coreAux** [reply]
Yes that was the exact issue I had.

✅ Will work
```
 verified domain: mydomain.com
 sender email: hello@mydomain.com
```

❌ Won't work
```
 verified domain: send.mydomain.com
 sender email: hello@mydomain.com
```

**Reply by MikeLundahl** [reply]
I just tried removing the subdomain. just like you illustrated. Still not working :( 

Maybe worth mentioning is that my domain is hosted at Netlify (not sure it matters)

**Reply by coreAux** [reply]
Even if it's through Netlify, if it get's verified, that shouldn't affect it.
If you get it verified that means you've added all the DNS records correctly so let's assume that's correct.

Did you also reconnect the SMTP settings between Resend and Supabase?

**Reply by MikeLundahl** [reply]
> Did you also reconnect the SMTP settings between Resend and Supabase?

Sorry for the late reply, but yes. I did this.

**Reply by coreAux** [reply]
No worries - sorry to hear you're still having issues. Unfortunately I can't help you any further. But if I were you, the final things I would try just to make sure would be to:
1. Change the ports to check if that could help the issue, 465 and 587 are the options and have to be changed on both Supabase and Resend.
2. I tried restarting my project as well, from what I could see it didn't do anything but I read people who tried restarting and that helped.
3. As a final one try disconnect everything, create a new project on Supabase and see if the connection works with that new project.

Good luck! 🍀
